lpwan-backend/src/main/java/com/cumulocity/lpwan/lns/connection/service/LnsConnectionService.java

```diff
@@ -330,7 +331,10 @@ private OptionPK getLnsConnectionsTenantOptionKey() {
     }
 
     private List<LpwanDevice> getDeviceMoListAssociatedWithLnsConnection(String lnsConnectionName) throws LpwanServiceException {
-        InventoryFilter inventoryFilter = LpwanDeviceFilter.of("lnsConnectionName", lnsConnectionName);
+        Map<String, String> filterCriteriaMap = new HashMap<>();
+        filterCriteriaMap.put("lnsConnectionName", lnsConnectionName);
+        filterCriteriaMap.put("serviceProvider", LnsConnectionDeserializer.getRegisteredAgentName());
+        InventoryFilter inventoryFilter = LpwanDeviceFilter.of(filterCriteriaMap);
```

Comment:
- [ ] replace this with 
`LpwanDeviceFilter.byServiceProviderAndLnsConnectionName(LnsConnectionDeserializer.getRegisteredAgentName(), lnsConnectionName)`

Reply:
This has been done now